### ISSUE TITLE
Error creating function with New-AzFunctionApp on PowerShell 5.1

### ISSUE THREAD

**Issue by DiscoveringWorld** [Azure/azure-powershell#15430]
Hello,

I got an issue with creating function with  New-AzFunctionApp on PowerShell 5.1 after updated to

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     3.1.0      Az.Functions                        {Get-AzFunctionApp, Get-AzFunctionAppAvailableLocation, Get-AzFunctionAppPlan, Get-AzFunctionAppSetting...}

```
$resLocation = 'USGov Virginia'
$CSPFunctionRGName = "CSPFunction12343424"
$CSPFunctionAppPlanName = "FunctionAppPlan123423"
$CSPFunctionAppName = "function123422322"
$CSPStorageFunctionName = "storagefunction123422322"

New-AzResourceGroup -Name $CSPFunctionRGName -Location $resLocation 
New-AzStorageAccount -ResourceGroupName $CSPFunctionRGName -AccountName $CSPStorageFunctionName -Location $resLocation -SkuName Standard_RAGRS 
New-AzFunctionAppPlan -ResourceGroupName $CSPFunctionRGName -Name $CSPFunctionAppPlanName -WorkerType Windows -Sku  EP1   -Location $resLocation 
New-AzFunctionApp -Name $CSPFunctionAppName   -ResourceGroupName $CSPFunctionRGName  -Location $resLocation  -Runtime dotnet -RuntimeVersion 3 -FunctionsVersion 3  -StorageAccount $CSPStorageFunctionName 


VERBOSE: OSType for dotnet is 'Windows'.
GetShareName : Cannot convert value "a" to type "System.Int32". Error: "Input string was not in a correct format."
At C:\Users\discoveringworld\Documents\WindowsPowerShell\Modules\Az.Functions\3.1.0\custom\New-AzFunctionApp.ps1:419 char:26
+             $shareName = GetShareName -FunctionAppName $Name
+                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [GetShareName], RuntimeException
    + FullyQualifiedErrorId : InvalidCastFromStringToInteger,GetShareName


PS C:\Windows\system32> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.18362.1593
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.1593
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by DiscoveringWorld** [comment]
I took bits of code from the powershell to understand the error and here is the error

```
$FunctionAppName = "function123422322"
$FunctionAppName = $FunctionAppName.ToLower()

# Share name will be function app name + 8 random char suffix with a max length of 60
   $MAXLENGTH = 60
   $SUFFIXLENGTH = 8
   if (($FunctionAppName.Length + $SUFFIXLENGTH) -lt $MAXLENGTH)
   {
       $name = $FunctionAppName
   }
   else
   {
       $endIndex = $MAXLENGTH - $SUFFIXLENGTH - 1
       $name = $FunctionAppName.Substring(0, $endIndex)
   }

   $suffix = GetShareSuffix -Length $SUFFIXLENGTH
   $shareName = $name + $suffix

Cannot convert value "a" to type "System.Int32". Error: "Input string was not in a correct format."
At line:9 char:5
+     $letters = 'a'..'z'
+     ~~~~~~~~~~~~~~~~~~~
   + CategoryInfo          : InvalidArgument: (:) [], RuntimeException
   + FullyQualifiedErrorId : InvalidCastFromStringToInteger


function GetShareSuffix
{
   [Microsoft.Azure.PowerShell.Cmdlets.Functions.DoNotExportAttribute()]
   param
   (
       [Int]
       $Length = 8
   )

   $letters = 'a'..'z'
   $numbers = 0..9
   $alphanumericLowerCase = $letters + $numbers

   $suffix = [System.Text.StringBuilder]::new()

   for ($index = 0; $index -lt $Length; $index++)
   {
       $value = $alphanumericLowerCase | Get-Random
       $suffix.Append($value) | Out-Null
   }

   $suffix.ToString()
}
```

**Comment by DiscoveringWorld** [comment]
If you don't want to fix that on  PowerShell 5.1 it's fine. We can close that.

I just use PowerShell 7.1.4 and later now.

**Comment by Francisco-Gamino** [comment]
This issue has been fixed.
